src/DataTypes/DataTypesNumber.cpp

```diff
@@ -2,23 +2,52 @@
 #include <DataTypes/DataTypeFactory.h>
 
 
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+
+
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int UNEXPECTED_AST_STRUCTURE;
+}
+
+template <typename T>
+static DataTypePtr createNumericDataType(const ASTPtr & arguments)
+{
+    if (arguments)
+    {
+        if (std::is_integral_v<T>)
+        {
+            if (arguments->children.size() > 1)
+                throw Exception(String(TypeName<T>::get()) + " data type family must not have more than one argument - display width", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        }
+        else
+        {
+            if (arguments->children.size() > 2)
+                throw Exception(String(TypeName<T>::get()) + " data type family must not have more than two arguments - total number of digits and number of digits following the decimal point", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        }
+    }
+    return std::make_shared<DataTypeNumber<T>>();
+}
+
+
 void registerDataTypeNumbers(DataTypeFactory & factory)
 {
     factory.registerSimpleDataType("UInt8", [] { return DataTypePtr(std::make_shared<DataTypeUInt8>()); });
```

Comment:
Maybe we should do the same thing for other numeric types?

Reply:
which types do you mean?

Reply:
UInt8, UInt16, UInt32, UInt64

Reply:
I mean it seems it would be nice to have syntax `UInt32(number)` for consistency.